app/models/article.rb

```diff
@@ -652,6 +654,10 @@ def validate_tag
     end
   end
 
+  def validate_title
+    remove_prohibitted_unicode_characters(str: title) if contains_prohibitted_unicode_characters?(str: title)
```

Comment:
Do we also need to set `self.title=` the returned title string from remove prohibited unicode characters? It looks like this might be called without using the returned value. I might have missed something and haven't had a chance to verify that yet.

Compare validate_tag above and add_tag_adjustments_to_tag_list below which are used to modify the model before save/update.

Validation methods normally should raise an error (it looks like we're handling the problem condition directly by replacement, which is a valid strategy which might not need any errors to be reported when the validation fails - since it never does).